### PR TITLE
Add author, creationDate, keywords, modifiedDate, subject & title to …

### PR DESCRIPTION
**Pull request by becoded**
Add author, creationDate, keywords, modifiedDate, subject & title to the document information dictionary

### CODE REVIEW

File: pdf/model/structures.go [hunk 1]
```diff
@@ -124,6 +125,25 @@ func NewPdfDate(dateStr string) (PdfDate, error) {
 	return d, nil
 }
 
+func NewPdfDateFromTime(timeObj time.Time) (PdfDate, error) {
```
Comment (adrg) [inline pdf/model/structures.go]: Could also be written as:

```
func NewPdfDateFromTime(timeObj time.Time) (PdfDate, error) {
	timezone := timeObj.Format("-07:00")
	utOffsetHours, _ := strconv.ParseInt(timezone[1:3], 10, 32)
	utOffsetMins, _ := strconv.ParseInt(timezone[4:6], 10, 32)

	return PdfDate{
		year:          int64(timeObj.Year()),
		month:         int64(timeObj.Month()),
		day:           int64(timeObj.Day()),
		hour:          int64(timeObj.Hour()),
		minute:        int64(timeObj.Minute()),
		second:        int64(timeObj.Second()),
		utOffsetSign:  timezone[0],
		utOffsetHours: utOffsetHours,
		utOffsetMins:  utOffsetMins,
	}, nil
}
```

File: pdf/model/writer.go [hunk 2]
```diff
@@ -93,8 +158,18 @@ func NewPdfWriter() PdfWriter {
 
 	// Creation info.
 	infoDict := MakeDict()
-	infoDict.Set("Producer", MakeString(getPdfProducer()))
```
Comment (adrg) [inline pdf/model/writer.go]: Could change the order of the statements for a bit more structure.
Also, maybe some of the fields should be added only if they are not empty.

```
	infoDict.Set("Producer", MakeString(getPdfProducer()))
	infoDict.Set("Creator", MakeString(getPdfCreator()))
        infoDict.Set("Author", MakeString(getPdfAuthor()))
	infoDict.Set("Subject", MakeString(getPdfSubject()))
	infoDict.Set("Title", MakeString(getPdfTitle()))
	infoDict.Set("Keywords", MakeString(getPdfKeywords()))

        // Set creation and modified dates.
	if cd, err := NewPdfDateFromTime(getPdfCreationDate()); err == nil {
		infoDict.Set("CreationDate", cd.ToPdfObject())
	}
	if cd, err := NewPdfDateFromTime(getPdfModifiedDate()); err == nil {
		infoDict.Set("ModDate", cd.ToPdfObject())
	}
```

File: pdf/model/writer.go [hunk 3]
```diff
@@ -46,6 +64,53 @@ func SetPdfCreator(creator string) {
...
+	return pdfModifiedDate
+}
+
+func SetPdfModifiedDate(modifiedDate time.Time) {
+	pdfCreationDate = modifiedDate
+}
+
+func getPdfProducer() string {
+	licenseKey := license.GetLicenseKey()
+
```
Comment (adrg) [inline pdf/model/writer.go]: Remove empty line.

File: pdf/model/writer.go [hunk 4]
```diff
@@ -26,11 +26,29 @@ import (
...
+var pdfTitle = ""
+
+func getPdfAuthor() string {
+	return pdfAuthor
+}
 
-func getPdfProducer() string {
-	licenseKey := license.GetLicenseKey()
-	return fmt.Sprintf("UniDoc v%s (%s) - http://unidoc.io", getUniDocVersion(), licenseKey.TypeToString())
+func SetPdfAuthor(author string) {
```
Comment (adrg) [inline pdf/model/writer.go]: The public functions should have documentation comments.

File: pdf/model/structures.go (hunk 1 above)
Comment (gunnsth) [reply pdf/model/structures.go]: https://github.com/unidoc/unidoc/wiki/UniDoc-Developer-Guide#12-doc-comments

File: pdf/model/structures.go [hunk 5]
```diff
@@ -124,6 +125,25 @@ func NewPdfDate(dateStr string) (PdfDate, error) {
...
+	timezone := timeObj.Format("-07:00")
+	d.utOffsetSign = []byte(timezone[0:1])[0]
+
+	d.utOffsetHours, _ = strconv.ParseInt(timezone[1:3], 10, 32)
+	d.utOffsetMins, _ = strconv.ParseInt(timezone[4:6], 10, 32)
+
+	return d, nil
+}
+
 // Convert to a PDF string object.
```
Comment (gunnsth) [inline pdf/model/structures.go]: https://github.com/unidoc/unidoc/wiki/UniDoc-Developer-Guide#12-doc-comments

**Comment by becoded** [comment]
Closed this PR in favor of https://github.com/unidoc/unidoc/pull/408

**Pull request closed without merging**